hips/hip-0007.md

```diff
@@ -0,0 +1,69 @@
+---
+hip: 7
+title: "Document and Track subprojects"
+authors: [ "Matt Farina <matt@mattfarina.com>" ]
+created: "2020-11-09"
+type: "process"
+status: "draft"
+---
+
+## Abstract
+
+There are numerous Helm subprojects where each can have zero or more Git
+repositories associated with them. This process provides a centralized method
+to document and track the subprojects, maintainers, and associated repos.
+
+## Motivation
+
+The current design uses a collection of documented and access controls to track
+this information while there are several gaps in the information. The Teams.md
+file attempted to track the different teams (i.e. subprojects, the org
+maintainers, etc). It is unclear what is a subproject, and how the org
+maintainers fit in from that list. The repositories owned by a subproject are
+not documented but are instead tracked via GitHub teams. The owners are
+scattered in various locations that are not consistent. In some cases they
+cannot be consistent (e.g. community management which does not have a GitHub
+repo). This has lead to confusion with the Org Maintainers and others.
+
+## Rationale
+
+Describe why particular design decisions were made.
+
+## Specification
+
+The source of truth will be a YAML file name [`subprojects.yaml`](../subprojects.yaml)
+stored at the root of the Helm community repository. It will have the following
+structure:
+
+```
+subprojects:
+- name: Helm Core
+  ownersLink: https://raw.githubusercontent.com/helm/helm/master/OWNERS
+  repos:
+  - https://github.com/helm/helm
+- name: Charts
+  owners:
+  - unguiculus
+  - ...
+  emeritus:
+  - foxish
+  - ...
+  repos:
+  - https://github.com/helm/chart-releaser-action
+  - https://github.com/helm/charts-repo-actions-demo
+  - ...
+```
+
+A subproject can only have either `ownersLink` (a URL to the location of the
+owners list) or `owners` (a list of owners). It cannot have both. This enables
+those who do not have a Git repo (e.g., community managers) and those with
+multiple repos and no primary one (e.g., charts) to have a documented list.
+
+When `owners` is used it can have an accompanying `emeritus`. When `ownersLink`
+is used the emeritus is expected to be listed in that file.
+
+## References
+
+* [Helm Governance](https://github.com/helm/community/blob/master/governance/governance.md)
```

Comment:
We may want to clarify the language in the governance doc as well - it references "subprojects", "sub-groups" and "subteam".

Reply:
I was looking at the governance. The item sub-groups means that the org maintainers create categories like "subprojects". It's not meant to refer to subprojects.

"subteam" should be fixed but I think that can be a different PR. It's something we didn't catch during the initial reviews of the governance.